src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1867,33 +1867,99 @@ private FlagsExpression<FileAttributes> FormatAttributeSwitchParameters()
         /// Provides a mode property for FileSystemInfo.
         /// </summary>
         /// <param name="instance">Instance of PSObject wrapping a FileSystemInfo.</param>
+        /// <returns>A string representation of the FileAttributes, with one letter per attribute.</returns>
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods")]
-        public static string Mode(PSObject instance)
+        public static string Mode(PSObject instance) => Mode(instance, excludeHardLink: false);
+
+        /// <summary>
+        /// Provides a ModeWithoutHardLink property for FileSystemInfo, without HardLinks for performance reasons.
+        /// </summary>
+        /// <param name="instance">Instance of PSObject wrapping a FileSystemInfo.</param>
+        /// <returns>A string representation of the FileAttributes, with one letter per attribute.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods")]
+        public static string ModeWithoutHardLink(PSObject instance) => Mode(instance, excludeHardLink: true);
+
+        private static string Mode(PSObject instance, bool excludeHardLink)
         {
-            if (instance == null)
+            string ToModeString(FileSystemInfo fileSystemInfo)
             {
-                return string.Empty;
-            }
+                FileAttributes fileAttributes = fileSystemInfo.Attributes;
 
-            FileSystemInfo fileInfo = (FileSystemInfo)instance.BaseObject;
-            if (fileInfo == null)
-            {
-                return string.Empty;
+                bool isReparsePoint = InternalSymbolicLinkLinkCodeMethods.IsReparsePoint(fileSystemInfo);
+                bool isLink = isReparsePoint || excludeHardLink ? false : InternalSymbolicLinkLinkCodeMethods.IsHardLink(fileSystemInfo);
+                if (!isLink)
+                {
+                    // special casing for the common cases - no allocations
+                    switch (fileAttributes)
+                    {
+                        case FileAttributes.Archive:
+                            return "-a---";
+                        case FileAttributes.Directory:
+                            return "d----";
+                        case FileAttributes.Normal:
+                            return "-----";
+                        case FileAttributes.Directory | FileAttributes.ReadOnly:
+                            return "d-r--";
+                        case FileAttributes.Archive | FileAttributes.ReadOnly:
+                            return "-ar--";
+                    }
+                }
+
+                bool isDirectory = fileAttributes.HasFlag(FileAttributes.Directory);
+                ReadOnlySpan<char> mode = stackalloc char[]
+                {
+                    isLink ? 'l' : isDirectory ? 'd' : '-',
```

Comment:
@powercode I changed it from `(isReparsePoint || isLink) ? 'l' ...` to `isLink ? 'l' ...` because we already have `bool isLink = isReparsePoint || ...` above, so `isReparsePoint` is not needed here again. Please let me know if you have a concern.

Reply:
LGTM

Reply:
Did that break the tests?

Reply:
> `bool isLink = isReparsePoint || excludeHardLink ? false : InternalSymbolicLinkLinkCodeMethods.IsHardLink(fileSystemInfo);`

The `||` operator has a higher precedence order than `?:`, so we need a parenthesis:
```
bool isLink = isReparsePoint || (excludeHardLink ? false : InternalSymbolicLinkLinkCodeMethods.IsHardLink(fileSystemInfo));
```

Reply:
Pushed a commit to fix that. Let's wait on the CIs.

Reply:
Your fix was good, but it exposed a presendense bug where 
```Csharp
isReparsePoint || excludeHardLink ? false : InternalSymbolicLinkLinkCodeMethods.IsHardLink(fileSystemInfo);
```
really is 
```csharp
(isReparsePoint || excludeHardLink) ? false : InternalSymbolicLinkLinkCodeMethods.IsHardLink(fileSystemInfo);
```
and not the intended

```csharp
isReparsePoint || (excludeHardLink ? false : InternalSymbolicLinkLinkCodeMethods.IsHardLink(fileSystemInfo));
```
pushed a fix.